librad/src/profile.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2019-2021 The Radicle Foundation <hello@radicle.foundation>
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{env, fs, io, path::Path};
+use uuid::Uuid;
+
+use crate::paths::{project_dirs, Paths};
+
+/// A [`Profile`] provides [`Paths`] scoped by an identifier.
+///
+/// Profiles with different identifiers have distinct paths.
+#[derive(Debug, Clone)]
+pub struct Profile {
+    id: String,
```

Comment:
Would there be any advantage to using the `Uuid` type here instead? Or even having a generic parameter for switching between different identity types in the future or for testing?

Reply:
I don’t see a strong need to make this generic. At the moment he tests don’t benefit from it and I can’t imagine that in the future we need a profile with two different ID parameters. If that happens though, it’s not to late to change the code.

Regarding using `Uuid` instead of `String`. If we implement this than only UUIDs are valid profile identifiers. This is a limitation that may make the feature harder to user. For example, say I’m playing around with this using the `RAD_PROFILE` environment variable. Since UUIDs are quite opaque, I’m free to use `RAD_PROFILE=foo` and radicle will deal with this. We only use `UUIDs` to generate unspecified profiles to ensure they are unique.

Reply:
Makes sense :+1: